practices/cicd/CICD.md

```diff
@@ -0,0 +1,53 @@
+# CI/CD
+
+- **Continuous Integration (CI)**: Automated integration of code
+- **Continuous Deployment (CD)**: Automated deployment of software
+
+## The What and the Why
+
+Proper CI/CD automates the manual stuff and ensures that code is merged and released correctly.
+- 🏎 Team velocity
+- 💻 Code accuracy
+- 🤓 Developer happiness 
+
+## CI
+
+CI is the task of regularly merging code changes. It allows for common tasks to be run to ensure code quality and integrity. Some common tasks/checks that happen at this stage are:
+
+- Testing (unit, integration, &/or end-to-end)
+- Linting (code format checks)
+- Building/compiling
+- Manual Code Reviews
+
+If any issues or errors come up, they can be addressed before merging code into higher branches or environments. CI systems can also **require** that these checks pass prior to proceeding.
+
+[What is Continuous Integration](https://www.youtube.com/watch?v=1er2cjUq1UI)
+
+## CD
+
+CD is the task of automating release and deployment of code to running environments. Common tasks/checks that happen at this stage are:
+
+- Testing (unit, integration, &/or end-to-end)
+- Linting (code format checks)
+- Building/compiling
+- Release versioning
```

Comment:
@adamschnaare is it worth bringing up the integrating of a system here? Not saying every deploy needs to spin up the entire ecosystem, but I generally see at least some end to end style testing with core components of the system

Reply:
I guess for the purposes of the doc I was leaving this system agnostic. Would that step into architecture thought? Maybe this doc expands and has ideal definitions of entire ci/cd flows for different tech setups/architectures?